### PR TITLE
Draw boat shadow (#1290)

### LINKED ISSUE #1290: Boat shadow is missing, while no hero is on board
Boat has no shadow, until any hero boards it.
Boat shadow

### PR DESCRIPTION
**Pull request by vasilenkoalexey**
Suggestion to fix #1290

### CODE REVIEW

**Review by ihhub — COMMENTED** [review]
Hi @vasilenkoalexey good job! Could you please take a look into few given comments?

File: src/fheroes2/maps/maps_tiles.cpp [hunk 1]
```diff
@@ -1570,13 +1570,17 @@ void Maps::Tiles::RedrawMonster( fheroes2::Image & dst ) const
...
+void Maps::Tiles::RedrawBoat( fheroes2::Image & dst, bool withShadow ) const
 {
     const Point mp = Maps::GetPoint( GetIndex() );
     const Interface::GameArea & area = Interface::Basic::Get().GetGameArea();
 
     if ( area.GetVisibleTileROI() & mp ) {
         const uint32_t spriteIndex = ( objectIndex == 255 ) ? 18 : objectIndex;
+        if ( withShadow ) {
+            const fheroes2::Sprite & sprite = fheroes2::AGG::GetICN( ICN::BOATSHAD, spriteIndex % 128 );
+            area.BlitOnTile( dst, sprite, sprite.x(), TILEWIDTH + sprite.y() - 11, mp, ( spriteIndex > 128 ) );
```
[the remark this answers is not in the document]
Comment (vasilenkoalexey) [reply src/fheroes2/maps/maps_tiles.cpp:1582]: Formally, there are two independent variables `sprite`, one for the boat, the second for the shadow, and theoretically they can have different `y()` value.

[the remark this answers is not in the document]
Comment (ihhub) [reply src/fheroes2/maps/maps_tiles.cpp:1582]: Ah, you're right. No need to change then.

File: src/fheroes2/maps/maps_tiles.h [hunk 2]
```diff
@@ -160,7 +160,7 @@ namespace Maps
         void RedrawTop( fheroes2::Image & dst ) const;
         void RedrawTop4Hero( fheroes2::Image &, bool skip_ground ) const;
         void RedrawObjects( fheroes2::Image & dst, bool isPuzzleDraw = false ) const;
-        void RedrawMonstersAndBoat( fheroes2::Image & ) const;
+        void RedrawMonstersAndBoat( fheroes2::Image &, bool withShadow = true ) const;
```
[the remark this answers is not in the document]
Comment (vasilenkoalexey) [reply src/fheroes2/maps/maps_tiles.h]: Done

File: src/fheroes2/maps/maps_tiles.h [hunk 3]
```diff
@@ -240,7 +240,7 @@ namespace Maps
         void RemoveJailSprite( void );
         bool isLongObject( int direction );
 
-        void RedrawBoat( fheroes2::Image & ) const;
+        void RedrawBoat( fheroes2::Image &, bool ) const;
```
[the remark this answers is not in the document]
Comment (vasilenkoalexey) [reply src/fheroes2/maps/maps_tiles.h]: Done

**Comment by ihhub** [comment]
@vasilenkoalexey thank you so much for fixing this issue!

**Pull request merged**